### PR TITLE
[WIP] Windows x86_64-v2 runners

### PR DESCRIPTION
**Pull request by jjnesbitt**

### CODE REVIEW

**Comment by mvandenburgh** [comment]
@AlmightyYakob fyi this branch will need to be rebased on `main` due to the EKS upgrade in #569.

**Comment by mvandenburgh** [comment]
@AlmightyYakob this branch will need to be rebased again on `main` due to infrastructure changes in #581 and #591.

**Comment by AlmightyYakob** [comment]
> @AlmightyYakob this branch will need to be rebased again on `main` due to infrastructure changes in #581 and #591. (just giving a heads up since `terraform apply` currently tries to delete those new resources when run on this branch)

This has been done.

File: k8s/production/runners/public/x86_64/v2/windows.yaml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      karpenter.sh/do-not-evict: true
+
+    runners:
+      # default image
+      image: "mcr.microsoft.com/windows/servercore:ltsc2022"
+      imagePullPolicy: "if-not-present"
+      locked: false
+
+      # tags: "windows"
+      tags: "spack,public,protected,small,medium,win64,kitware,x86_64,x86_64_v2,aws"
```
Comment (kwryankrattiger) [inline k8s/production/runners/public/x86_64/v2/windows.yaml:58]: keep `win64,`
drop `kitware`
mod `x86_64*` -> `win-x86_64*`

Comment (kwryankrattiger) [reply k8s/production/runners/public/x86_64/v2/windows.yaml:58]: public and protected shouldn't be the same

File: k8s/production/runners/public/x86_64/v2/windows.yaml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        kind: HelmRepository
+        name: runner-x86-v2-pub-windows
+  values:
+    image:
+      image: gitlab-org/gitlab-runner
+      tag: alpine-v16.1.0
+    imagePullPolicy: IfNotPresent
+    replicas: 1
+
+    gitlabUrl: "https://gitlab.staging.spack.io/"
```
Comment (kwryankrattiger) [inline k8s/production/runners/public/x86_64/v2/windows.yaml:33]: Prod should not be staging. For initial deployment only deploy this to staging for development

File: k8s/production/runners/public/x86_64/v2/windows.yaml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+kind: HelmRelease
+metadata:
+  name: runner-x86-v2-pub-windows
+  namespace: gitlab
+spec:
+  interval: 10m
+  chart:
+    spec:
+      chart: gitlab-runner
+      version: 0.54.0 # gitlab-runner@v16.1.0
```
Comment (kwryankrattiger) [inline k8s/production/runners/public/x86_64/v2/windows.yaml:22]: New runner

**Comment by jjnesbitt** [comment]
Superseded by #764

**Pull request closed without merging**